### PR TITLE
Updated versions of various Jetbrains applications, including PHPStor…

### PR DESCRIPTION
**Pull request by joncojonathan**
Update versions for some Jetbrains products, including PHPStorm, IntelliJ and PyCharm (non-exhaustive list).

###### Motivation for this change
Version of PHPStorm and IntelliJ I installed was out of date - contributing a correction.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
   - Tested pycharm-community, phpstorm and idea-community
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Hi, we usually have one commit per package update in the form: `packagename: <from_version> -> <to_version>`.

**Comment by joncojonathan** [comment]
Hi @Mic92 , Sorry!  I'll remember that in future.  Is that a problem in this case?

New to this, so expecting to learn as I go.

**Comment by Mic92** [comment]
In case of intellij it might be acceptable to upgrade in one commit as all components carry the same version. But please change the commit message to:

```
idea: 2016.3.2 -> 2016.3.3
```

File: pkgs/applications/editors/idea/default.nix [hunk 1]
```diff
@@ -340,12 +340,12 @@ in
 
   datagrip = buildDataGrip rec {
     name = "datagrip-${version}";
-    version = "2016.3.2";
+    version = "2016.3";
```
Comment (globin) [inline pkgs/applications/editors/idea/default.nix]: That looks like a downgrade, was that intentional?

Comment (joncojonathan) [inline pkgs/applications/editors/idea/default.nix]: I didn't edit that part, not intentionally anyway.

Comment (joncojonathan) [inline pkgs/applications/editors/idea/default.nix]: @globin how do we fix the potential downgrade, do I need to perform a full recommit and make a new pull request?

**Comment by joncojonathan** [comment]
@Mic92 versions now specified in the commit message.

File: pkgs/applications/editors/idea/default.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/applications/editors/idea/default.nix]: no. leave this pull request open. You can added changes to your existing commit using `git commit --amend` and a `git push --force origin update-jetbrains`

Comment (joncojonathan) [inline pkgs/applications/editors/idea/default.nix]: @globin @Mic92 I've updated the commit message and corrected the datagrip version.  I've also corrected the hash for it as that seems to have regressed.

I've tested datagrip too.

Comment (globin) [inline pkgs/applications/editors/idea/default.nix]: You don't seem to have pushed the fix to the downgrade, might need to git add before amending the commit

**Comment by joncojonathan** [comment]
@Mic92 @globin I think that's everything ironed out now.  I've also added the remaining jetbrains packages.

Please let me know if you require anything further?

**Comment by grahamc** [comment]
This looks fine to me, but do note that the commit message is definitely deviant from the norm. In the future, please upgrade one package per commit and format the first line of the message "foo: 0.0.0 -> 0.0.1"

**Comment by grahamc** [comment]
Oh, I meant to say, thank you so much for the patches :) We definitely appreciate them!

**Comment by joncojonathan** [comment]
@grahamc No problem, and I've noted the commit message style for future.

**Pull request merged**